components/session-service/server/server.go

```diff
@@ -347,15 +352,89 @@ func (s *Server) logoutHandler(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func filterUserPolicy(policyId string, policyIds []string) bool {
+	for _, _policyId := range policyIds {
+		if _policyId == policyId {
+			return true
+		}
+	}
+	return false
+}
+
 func (s *Server) userPoliciesHandler(w http.ResponseWriter, r *http.Request) {
 	decoder := json.NewDecoder(r.Body)
 	var data userData
 	err := decoder.Decode(&data)
 	if err != nil {
-		fmt.Println(err, "error")
+		http.Error(w, errors.Wrap(err, "failed to decode").Error(),
+			http.StatusInternalServerError)
+		return
+	}
+
+	teamsForUser, err := s.teamsServiceClient.GetTeamsForMember(r.Context(), &teams.GetTeamsForMemberReq{
+		UserId: data.UserId,
+	})
+	if err != nil {
+		http.Error(w, errors.Wrap(err, "failed teamsServiceClient").Error(),
+			http.StatusInternalServerError)
+		return
+	}
+
+	var teamsPolicyIds []string
+	if len(teamsForUser.Teams) > 0 {
+		for _, team := range teamsForUser.Teams {
+			teamPolicies, err := s.authzPoliciesClient.GetUserPolicies(r.Context(), &authz.GetUserPoliciesReq{
```

Comment:
here we are fetching the policies for each team, instead of this can we fetch them all in one call by sending the list?

Reply:
code updated